### DISCUSSION TITLE
How to edit source code with custom block.

### DISCUSSION THREAD

**Discussion by daviddanielng** [Sketchware-Pro/Sketchware-Pro#457] [Q&A]
Hi everyone.
I downloaded a custom block from arabware, the block enabled replacing of parts of source code with my choice of code . I have looked at the block and I found out it is something different from how I know blocks are created.
It is something like this 👇👇👇
```
/*-JX4UA2y_f1OckjjvxWI.bQwRei-sLEsBmds7ArsRfi0xSFEP3Php97kjdMCs5ed>["public` class %2$s extends Activity {"]
>0
>0
>0
>find-replace
public class %2$s extends Activity implements %1$s {
BpWI8U4flOpx8Ke66QTlZYBA_NEusQ7BN-D0wvZs7ArsRfi0.EP3Php97kjdMCs*/
/*-JX4UA2y_f1OckjjvxWI.bQwRei-sLEsBmds7ArsRfi0xSFEP3Php97kjdMCs5ed
>["public class %2$s extends AppCompatActivity {"]
>0
>0
>0
>find-replace
public class %2$s extends AppCompatActivity implements %1$s {
BpWI8U4flOpx8Ke66QTlZYBA_NEusQ7BN-D0wvZs7ArsRfi0.EP3Php97kjdMCs*/
```

Can anyone tell me how to do this. I have search the internet but did not find any information.

**Comment by j-px** [comment]
That's just a Java Command Block. It and also the XML Command Block are available in Sketchware Pro v6.3.0 and newer. You can find them in the More Blocks palette like here:

![Screenshot_20220617-112115_Sketchware Pro](https://user-images.githubusercontent.com/40118124/174269558-5f76273e-2541-4d5c-8aac-e652090eab93.png)

after enabling this Mod Setting: (screenshot from v6.4.0, differently labeled in v6.3.0)

![Screenshot_20220617-112127_Sketchware Pro](https://user-images.githubusercontent.com/40118124/174269662-01852f28-01ef-4bd8-a7bc-1d45a94d4327.png)

**Reply by j-px** [reply]
And also, if you need a tutorial on them, the #tips-n-tricks channel in the official Discord server has some. Invite link to the server is available in this repo's README.
